dashboard/test/models/workshop_test.rb

```diff
@@ -1,7 +1,124 @@
 require 'test_helper'
 
 class WorkshopTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  # see also tests in OpsMailerTest
+
+  def create_workshop(segment_starts_and_ends)
+    create(:workshop).tap do |workshop|
+      next if segment_starts_and_ends.blank?
+
+      workshop.segments = []
+      segment_starts_and_ends.each do |segment_start_and_end|
+        workshop.segments.build(start: segment_start_and_end.first,
+                                end: segment_start_and_end.last)
+      end
+      workshop.save!
+
+    end
+  end
+
+  setup do
+    @old_workshop = create_workshop [[Time.now.utc - 10.days, Time.now.utc - 9.days]]
+    @tomorrow_workshop = create_workshop [[Time.now.utc + 1.days, Time.now.utc + 1.days + 1.hour]]
+
+    now = Date.today
```

Comment:
Rails/Date: Do not use `Date.today` without zone. Use `Time.zone.today` instead.